### PR TITLE
ComposeBox: Move compose buttons into a row below message-text input

### PR DESCRIPTION
**Pull request by chrisbobbe**
Making `multiline` constantly true fixes various urgent bugs where
the message-text input would lose its contents or fail to update
when you press one of these menu buttons on iOS. Details in the
linked issues.

Fixes: #5080
Fixes: #5291
Fixes: #5463
Fixes: #4628

-----

cc @alya for the UX change

Before (notice my message "Qwerty" disappearing—I'm glad it was so short! 🥲):

After:

### CODE REVIEW

**Comment by chrisbobbe** [comment]
Fixing this will unblock my fixes for 

- #2366
- #4540

in my PR #5474.

**Comment by alya** [comment]
Nice! Can we make the buttons lighter in dark theme? They seem hard to see in the screenshot above.

Otherwise looks good to me!

**Comment by chrisbobbe** [comment]
Sure. I've made them a bit darker in light theme too, for good measure:

 

In `main` they are the blue "brand" color; I thought that was a bit too distracting with the new layout so I made them gray, but if you think the brand color might be better I can send screenshots of it.

**Comment by alya** [comment]
Cool, that looks reasonable to me. Thanks! I agree that brand color doesn't seem necessary here.

File: src/compose/ComposeBox.js [hunk 1]
```diff
@@ -199,10 +199,6 @@ const ComposeBox: React$AbstractComponent<Props, ImperativeHandle> = forwardRef(
 
   const inputBlurTimeoutId = useRef<?TimeoutID>(null);
 
-  // TODO(#5141): Encapsulate this in a nice, plain action-sheet pattern
-  //   instead of setting it all over the place
```
Comment (gnprice) [inline src/compose/ComposeBox.js:203]: This gives us a reminder that we should close #5141 — basically it's an alternative design that's mutually exclusive with #5080.

File: src/compose/ComposeMenu.js [hunk 2]
```diff
@@ -212,48 +232,28 @@ export default function ComposeMenu(props: Props): Node {
...
-          padding: 12,
-          color: BRAND_COLOR,
-        },
-        composeMenuButton: {
-          padding: 12,
-          marginRight: -8,
-          color: BRAND_COLOR,
+          backgroundColor,
         },
       }),
```
Comment (gnprice) [inline src/compose/ComposeMenu.js:243]: image

I want to iterate a bit more on the design before we merge this. In particular, that thin horizontal line between the content box and the row of icons is jumping out at me — it gives me the feeling that something's glitched here.

Let's try leaving out this `backgroundColor` prop. Then we get something like the following:
tmp
which I think looks a lot more plausible.

Two other things that look off to me when that `backgroundColor` is there:
* The row of icons looks a lot like the two text inputs — so it looks like it wants to be a third text input. But in fact it's quite different from a text input.
* There's a bunch of empty-looking space off to the right, where the background color keeps going past the icons it's there for.

File: src/compose/ComposeMenu.js [hunk 3]
```diff
@@ -212,48 +232,28 @@ export default function ComposeMenu(props: Props): Node {
   const styles = useMemo(
     () =>
       createStyleSheet({
-        composeMenu: {
+        container: {
+          marginTop: 2, // So it doesn't run into the text input
```
Comment (gnprice) [inline src/compose/ComposeMenu.js:236]: Let's put more spacing here. Say, 4px. (That's what you see in my screenshot in the previous comment.)

File: src/compose/ComposeBox.js [hunk 4]
```diff
@@ -684,8 +658,13 @@ const ComposeBox: React$AbstractComponent<Props, ImperativeHandle> = forwardRef(
           ...(!canSelectTopic && { position: 'absolute', transform: [{ scale: 0 }] }),
         },
         composeTextInput: {
+          // These border attributes override styles set in <Input />.
           borderWidth: 0,
-          borderRadius: 5,
+          borderTopLeftRadius: 5,
+          borderTopRightRadius: 5,
+          borderBottomLeftRadius: 0,
+          borderBottomRightRadius: 0,
```
Comment (gnprice) [inline src/compose/ComposeBox.js:666]: (This becomes unnecessary when there's no longer a special background for the icons row, per my previous comment.)

**Review by gnprice — COMMENTED** [review]
Thanks @chrisbobbe! This code looks good, but I'd like to iterate a bit more on the design before we merge this. Details below.

File: src/compose/ComposeMenu.js (hunk 2 above)
Comment (chrisbobbe) [reply src/compose/ComposeMenu.js:243]: Sure, makes sense. I was looking at an old screenshot from Vlad on [CZO](https://chat.zulip.org/#narrow/stream/48-mobile/topic/UI.20redesign.3A.20Compose.20buttons/near/1472337), and trying to move slightly in the direction of the design there—while avoiding:
- complication in the diff
- having to pick a new color for the background of the buttons panel



Your proposal successfully avoids those things, and I think also doesn't really get in the way of following Vlad's design in that screenshot (which was preliminary anyway and may change; [Vlad plans to open an issue](https://chat.zulip.org/#narrow/stream/48-mobile/topic/UI.20redesign.3A.20Compose.20buttons/near/1481618)). 🙂

File: src/compose/ComposeMenu.js (hunk 3 above)
Comment (chrisbobbe) [reply src/compose/ComposeMenu.js:236]: Hmm, with these buttons not being on their own panel, do you think they want to be vertically centered between the bottom of the message input and the bottom of the compose box? Here's with 4px:

image

How about 8px:

image

Then if that seems too loose and we want a denser layout, one idea is to change that and all the other 8px padding/margins in the compose box to be 4px instead (which, it's hard to tell, might go in the direction of Vlad's screenshot?):

image

Another is to take the 8px distance below and above the bottons(' touchable areas) and just make it zero:

image

File: src/compose/ComposeBox.js (hunk 1 above)
Comment (chrisbobbe) [reply src/compose/ComposeBox.js:203]: Sure; I can remember to do that after this is merged.

**Comment by chrisbobbe** [comment]
Thanks for the review! Revision pushed; here are the new screenshots:

 

Ah, and one with the keyboard down, and one with touch targets highlighted:

**Comment by gnprice** [comment]
Thanks for the revision! This looks good — merging.

**Pull request merged**